apps/ospedale/src/Ospedale.purs

```diff
@@ -0,0 +1,148 @@
+        newArticle :: Effect Unit
+        newArticle = do
+          Routes.setRoute nav "/artikel"
+    useEffectOnce do
+      case initialSession of
+        Nothing -> pure $ pure unit
+        Just ini -> do
+          fiber <- Aff.launchAff do
+            maybeSession <- ini
+            liftEffect do
+              setSession maybeSession
+              setLoading Nothing
+          pure $ Aff.launchAff_ $ cancel fiber
+
+    useEffectOnce do
+      case initialRoute of
+        Routes.ArticlePage _ -> Routes.setRoute nav "/"
+        _ -> pure unit
+
+      Routes.routeListener nav $ \r -> do
+        setRoute r
+
+    let doPurge purge tok = do
+          setPurgeResult Nothing
+          Aff.launchAff_ $ Spinner.withSpinner setLoading do
+            liftEffect <<< setPurgeResult <<< Just =<< Lettera.prerenderedCachePurge tok purge
+
+    let renderSite accessToken = fold
+          [ DOM.div
+              { children:
+                  [ DOM.button
+                      { children: [ DOM.text "Purge prerendered frontpage cache" ]
+                      , onClick: foldMap (capture_ <<< doPurge Lettera.Purge) accessToken
+                      }
+                  , DOM.button
+                      { children: [ DOM.text "Enable using prerendered frontpage" ]
+                      , onClick: foldMap (capture_ <<< doPurge Lettera.Reenable) accessToken
+                      }
+                  , flip foldMap purgeResult $ \result -> case result of
+                        Right _ -> DOM.text "OK"
+                        Left err -> renderError relogin "Logga in igen" err
+                  ]
+              }
+          , DOM.div
+              { className: "ksf-main-container"
+              , children: pure $
+                  let list = articleComponent
+                        { openArticle
+                        , setNavPaper: setPaper <<< Just
+                        , newArticle
+                        , closeArticle: Routes.setRoute nav "/"
+                        , initialArticle:
+                          case initialRoute of
+                            Routes.ArticlePage a -> a
+                            _ -> Nothing
+                        , relogin
+                        , token: accessToken
+                        }
+                  in case route of
+                    Routes.MissingPage -> renderMissing $ Routes.setRoute nav "/"
+                    _ -> list
+              }
+          ]
+
+    pure $ fold
+      [ navbar
+          { paper: fromMaybe Paper.KSF paper
+          , activeUser: _.email <$> session
+          , specialHelp: Nothing
+          , logout
+          }
+      , case {loading, session} of
+            {loading: Just _} -> Spinner.loadingSpinner
+            {session: Nothing} -> loginComponent {setSession: setSession <<< Just}
+            {session: Just s} -> renderSite (Just s.session)
+      ]
+
+jsApp :: {} -> JSX
+jsApp = unsafePerformEffect app
```

Comment:
I don't know about PureScript, but I think that `unsafePerformIO` is generally regarded as an antipattern in Haskell.